active/0000-nested-addon-imports.md

```diff
@@ -0,0 +1,63 @@
+- Start Date: (fill me in with today's date, 2016-08-06)
+- RFC PR: 
+- ember-cli Issue: 
+
+# Summary
+
+The new asset import API, available since ember-cli 2.7.0, offers the possibility to 
+import Ember CLI addons from Ember CLI addons (aka, nested addons).
+This RFC intends to find solutions for typical edge cases.
+
+# Motivation
+
+Since ember-cli 2.7.0, addons can be nested if the nesting floows the following
+scenario:
+* The nested addons do not make use of the property `parent.bowerProperty` to define
+  assets paths because it not always defined (it is not defined in EmberAddon instances)
+* The nested addons do not need to execute a default blueprint.
+
+Solving these use cases would allow for a simpler, more generic, addons development
+process, and will allow teams to build more reusable and higher level addons for
+their projects and the community (like [ember-admin](https://github.com/DockYard/ember-admin)
+using [ember-power-select](ember-power-select.com) with no additional boilerplate, for instance).
+
+# Detailed design
+
+Here is a rundown of the questions raised with @nathanhammond in [a previous PR](https://github.com/ember-cli/ember-cli/pull/6111):
+
+1. Should nested addons be allowed to use their grandparents' `bower_components` directory?
+1. How/When should the default blueprints of the nested be run?
+1. How to deal with conflicting versions (The app uses my-addon 1.0.0, but my-addon 2.0.0 is required as a nested component)?
+
+For each of these questions, here's a first attempt towards an answer:
+
+1. Yes, they should, because there's only one bower directory available to the app, and it should be available to the whole app,
+  including to the dependencies. See [the PR linked above](https://github.com/ember-cli/ember-cli/pull/6111) for an example.
+1. They could be run in the "afterInstall" hook, when the dependencies (hence nested addons) are already installed.
+  See [this comment](https://github.com/ember-cli/ember-cli/pull/6111#issuecomment-235829717) for a specific implementation
+1. Ideally, this falls under Bower's (or npm's) umbrella.
```

Comment:
Right now we've got a solution where the "last one wins" as it simply does a `broccoli-merge` of the directories. Figuring out which one is last is actually an `undefined` implementation detail, however. It's lexicographically sorted, but also depth sorted. How we should support this remains unclear to me.

I think we'd need a module isolation per addon ability to make this work and rewrite `import`s in the addon to point to a particular version if we're going to support Node-style module scoping instead of a single universal module scope.